Spurious CA1305 for Guid.Parse with latest-recommended analysis level in net7.0

### Description

After updating the `TargetFramework` for an existing program that uses `<AnalysisLevel>latest-recommended</AnalysisLevel>` to `net7.0`, analysis starts generating spurious CA1305 warnings for `Guid.Parse`, because it now has an overload that takes `IFormatProvider` (but completely ignores it) https://github.com/dotnet/runtime/blob/af2a48ddaec9a3e27002b0d3b80c07431c2e38c1/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L1490

If `TreatWarningsAsErrors` is `true`, the program no longer compiles.

### Reproduction Steps

Create a new Console app with this csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <AnalysisLevel>latest-recommended</AnalysisLevel>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>

</Project>
```

(This is basically adding `<AnalysisLevel>latest-recommended</AnalysisLevel><TreatWarningsAsErrors>true</TreatWarningsAsErrors>` to the results of `dotnet new console`.

And this source code:

```csharp
Console.WriteLine(Guid.Parse("11223344-5566-7788-99AA-BBCCDDEEFF00"));
```

Run `dotnet run`:

> C:\Code\Projects\GuidParse\Program.cs(1,19): error CA1305: The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings. Replace this call in '<top-level-statements-entry-point>' with a call to 'Guid.Parse(string, IFormatProvider)'. [C:\Code\Projects\GuidParse\GuidParse.csproj]

If it's changed back to `<TargetFramework>net6.0</TargetFramework>`, the program compiles and runs normally.

### Expected behavior

No CA1305 analysis warning is introduced for existing working code when updating the target framework to `net7.0`, particularly because no benefit is gained by using the suggested overload that takes an `IFormatProvider`.

### Actual behavior

`dotnet run` produces this output and doesn't run the program:

> C:\Code\Projects\GuidParse\Program.cs(1,19): error CA1305: The behavior of 'Guid.Parse(string)' could vary based on the current user's locale settings. Replace this call in '<top-level-statements-entry-point>' with a call to 'Guid.Parse(string, IFormatProvider)'. [C:\Code\Projects\GuidParse\GuidParse.csproj]


### Regression?

Yes; no problem with this code targeting `net6.0`.

### Known Workarounds

_No response_

### Configuration

7.0.100-preview.4.22252.9
Windows 10, x64

### Other information

_No response_

--- a/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs
+++ b/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs
@@ -147,7 +147,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime
                 targetMethod.ContainingType.IsErrorType() ||
                 (activatorType != null && activatorType.Equals(targetMethod.ContainingType)) ||
                 (resourceManagerType != null && resourceManagerType.Equals(targetMethod.ContainingType)) ||
-                IsValidToStringCall(invocationExpression, invariantToStringTypes, dateTimeType, dateTimeOffsetType, timeSpanType))
+                IsValidToStringCall(invocationExpression, invariantToStringTypes, dateTimeType, dateTimeOffsetType, timeSpanType) ||
+                IsValidParseCall(invocationExpression, guidType))
                 {
                     return;
                 }
@@ -308,6 +309,23 @@ namespace Microsoft.NetCore.Analyzers.Runtime
                     return nullableTypeArgument;
                 return typeSymbol;
             }
+        }
+
+        private static bool IsValidParseCall(IInvocationOperation invocationOperation, INamedTypeSymbol? guidType)
+        {
+            var targetMethod = invocationOperation.TargetMethod;
+
+            if (targetMethod.Name != "Parse")
+            {
+                return false;
+            }
+
+            if (guidType != null && targetMethod.ContainingType.Equals(guidType))
+            {
+                return false;
+            }
+
+            return false;
         }
     }
 }

--- a/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs
+++ b/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs
@@ -322,7 +322,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime
 
             if (guidType != null && targetMethod.ContainingType.Equals(guidType))
             {
-                return false;
+                return true;
             }
 
             return false;

--- a/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs
+++ b/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs
@@ -320,7 +320,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime
                 return false;
             }
 
-            if (guidType != null && targetMethod.ContainingType.Equals(guidType))
+            if (targetMethod.ContainingType.Equals(guidType))
             {
                 return true;
             }

--- a/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProviderTests.cs
+++ b/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProviderTests.cs
@@ -946,6 +946,45 @@ End Class
         }
 
         [Fact]
+        [WorkItem(5999, "https://github.com/dotnet/roslyn-analyzers/issues/5999")]
+        public async Task CA1305_GuidParse_NoDiagnosticsAsync()
+        {
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = new ReferenceAssemblies(""), // workaround for lack of .NET 7 Preview 4 reference assemblies
+                TestCode = @"
+namespace System
+{
+    public class Object { }
+    public abstract class ValueType { }
+    public struct Void { }
+    public class String { }
+    public interface IFormatProvider { }
+    public struct Guid
+    {
+        public static Guid Parse(string s) => default;
+        public static Guid Parse(string s, IFormatProvider provider) => default;
+    }
+}
+namespace System.Globalization
+{
+    public class CultureInfo : IFormatProvider { }
+}
+namespace Test
+{
+    using System;
+    public class SomeClass
+    {
+        public Guid SomeMethod(string s)
+        {
+            return Guid.Parse(s);
+        }
+    }
+}",
+            }.RunAsync();
+        }
+
+        [Fact]
         public async Task CA1305_NullableInvariantTypes_NoDiagnosticAsync()
         {
             await VerifyCS.VerifyAnalyzerAsync(@"

